api/profile/controllers/profile.js

```diff
@@ -56,4 +56,17 @@ module.exports = {
     }
     return sanitizeEntity(entity, { model: strapi.models.profile });
   },
+  async delete(ctx) {
+    const { id } = ctx.params;
+    const { user } = ctx.state;
+    if (user.profile === parseInt(id)) {
+      if (user.role.name !== 'Teacher') {
+        await strapi.plugins['users-permissions'].services.user.remove({ id: user.id });
+        const entity = await strapi.services.profile.delete({ id });
+        return sanitizeEntity(entity, { model: strapi.models.profile });
+      }
+      return ctx.unauthorized(`You can't delete a user with role teacher`);
+    }
+    return ctx.unauthorized(`You can't delete this entry`);
+  },
```

Comment:
I think we do not need to pass an `id` as a parameter here since a user can only delete their own profile, we just use the user from JWT `const { user } = ctx.state;` and that's it, so we will end up with one if statement for the user role.

Reply:
I checked it & the user can delete any profile

Reply:
I do not think you can delete any profile, what I meant is something like this 
```
async delete (ctx) {
  const { user } = ctx.state; // <-- Current logged in user
  try {
    const { role } = user;
    if (role.type === 'teacher') return ctx.unauthorized(`You can't delete a user with role teacher`);
    
    await strapi.plugins['users-permissions'].services.user.remove({
      id: user.id,
    });
    const entity = await strapi.services.profile.delete({ id: user.profile });
    return sanitizeEntity(entity, { model: strapi.models.profile });
  } catch (e) {
    // handle me wa keda
  }
};
```

its basically the same logic but with less if statement and less nested conditions, right ?

Reply:
Yup..thanks